From: Saeed Mahameed <saeedm@mellanox.com>
Date: Thu, 10 Dec 2015 17:12:37 +0200
Subject: net/mlx5: Fix query E-Switch capabilities
Patch-mainline: v4.5-rc1
Git-commit: 9bd0a185c2c48226a50e6e5b639edaa9e8ab6c51
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

E-Switch capabilities should be queried only if E-Switch flow table
is supported and not only when vport group manager.

Fixes: d6666753c6e8 ("net/mlx5: E-Switch, Introduce HCA cap and E-Switch vport context")
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/fw.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/fw.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/fw.c
@@ -173,7 +173,7 @@ int mlx5_query_hca_caps(struct mlx5_core
 			return err;
 	}
 
-	if (MLX5_CAP_GEN(dev, vport_group_manager)) {
+	if (MLX5_CAP_GEN(dev, eswitch_flow_table)) {
 		err = mlx5_core_get_caps(dev, MLX5_CAP_ESWITCH,
 					 HCA_CAP_OPMOD_GET_CUR);
 		if (err)
